### ISSUE TITLE
[low priority] reduce debug output on clean cmake builds

### ISSUE THREAD

**Issue by mattklein123** [bazelbuild/rules_foreign_cc#216]
Here is a clean build for Envoy with the new foreign cmake rules:

It would be great if in the common clean case there was no/minimal debug output, but additional output could be required via a debug flag.

cc @htuch

```
/opt/slickedit-pro2018/bin/vsbuild -signal 43437 -command build_envoy.sh
 VSLICKERRORPATH=/home/mklein/Source/envoy
build_envoy.sh 
WARNING: The following rc files are no longer being read, please transfer their contents or import their path into one of the standard rc files:
/home/mklein/Source/envoy/tools/bazel.rc
INFO: Invocation ID: 54d8e913-3007-4d83-8253-256f4172cdec
INFO: Reading 'startup' options from /home/mklein/Source/envoy/.bazelrc: --host_jvm_args=-Xmx512m
INFO: Options provided by the client:
  Inherited 'common' options: --isatty=0 --terminal_columns=80
INFO: Reading rc options for 'build' from /home/mklein/Source/envoy/.bazelrc:
  'build' options: --workspace_status_command=bazel/get_workspace_status
INFO: Reading rc options for 'build' from /home/mklein/.bazelrc:
  'build' options: --action_env=CXX=clang++-7 --action_env=CC=clang-7 --experimental_strict_action_env=true --linkopt=-fuse-ld=lld-7 --announce_rc
Loading: 
Loading: 0 packages loaded
Analyzing: 9 targets (0 packages loaded, 0 targets configured)
DEBUG: /home/mklein/.cache/bazel/_bazel_mklein/0eed830f48870bc411befee27d7940f5/external/foreign_cc_impl/framework.bzl:269:5: script text: function cleanup_function() {
  local ecode=$?
  if [ $ecode -eq 0 ]; then
  printf "$CLEANUP_MSG"
  rm -rf $BUILD_TMPDIR $EXT_BUILD_DEPS
  else
  printf ""
  printf "$KEEP_MSG"
  printf ""
  fi
}
function children_to_path() {
  if [ -d $EXT_BUILD_DEPS/bin ]; then
    local tools=$(find $EXT_BUILD_DEPS/bin -maxdepth 1 -mindepth 1)
    for tool in $tools;
    do
      if  [[ -d "$tool" ]] || [[ -L "$tool" ]]; then
        export PATH=$PATH:$tool
      fi
    done
  fi
}
function replace_in_files() {
  if [ -d "$1" ]; then
    find -L $1 -print -type f   \( -name "*.pc" -or -name "*.la" -or -name "*-config" -or -name "*.cmake" \)   -exec sed -i 's@'"$2"'@'"$3"'@g' {} ';'
  fi
}
printf ""
printf "Bazel external C/C++ Rules #0.0.6. Building library 'event'"
printf ""
set -e

export EXT_BUILD_ROOT=$(pwd)
export BUILD_TMPDIR=$(mktemp -d)
export EXT_BUILD_DEPS=$EXT_BUILD_ROOT/bazel_foreign_cc_deps_event
export INSTALLDIR=$EXT_BUILD_ROOT/bazel-out/k8-fastbuild/genfiles/bazel/foreign_cc/event
export PATH="$EXT_BUILD_ROOT:$PATH"
mkdir -p $EXT_BUILD_DEPS
mkdir -p $INSTALLDIR
printf "Environment:______________"
env
printf "__________________________"
export CLEANUP_MSG="rules_foreign_cc: Cleaning temp directories"
export KEEP_MSG="rules_foreign_cc: Keeping temp build directory $BUILD_TMPDIR and dependencies directory $EXT_BUILD_DEPS for debug.\nrules_foreign_cc: Please note that the directories inside a sandbox are still cleaned unless you specify '--sandbox_debug' Bazel command line flag."
trap "cleanup_function" EXIT
children_to_path $EXT_BUILD_DEPS/bin
export PATH="$EXT_BUILD_DEPS/bin:$PATH"
replace_in_files $EXT_BUILD_DEPS BAZEL_GEN_ROOT $EXT_BUILD_DEPS
cd $BUILD_TMPDIR
export INSTALL_PREFIX="./event"
CC="/home/mklein/.cache/bazel/_bazel_mklein/0eed830f48870bc411befee27d7940f5/external/local_config_cc/extra_tools/envoy_cc_wrapper" CXX="/home/mklein/.cache/bazel/_bazel_mklein/0eed830f48870bc411befee27d7940f5/external/local_config_cc/extra_tools/envoy_cc_wrapper" CFLAGS="-U_FORTIFY_SOURCE -fstack-protector -Wall -Wthread-safety -Wself-assign -fcolor-diagnostics -fno-omit-frame-pointer -no-canonical-prefixes -Wno-builtin-macro-redefined -D__DATE__=\"redacted\" -D__TIMESTAMP__=\"redacted\" -D__TIME__=\"redacted\"" CXXFLAGS="-U_FORTIFY_SOURCE -fstack-protector -Wall -Wthread-safety -Wself-assign -fcolor-diagnostics -fno-omit-frame-pointer -std=c++0x -no-canonical-prefixes -Wno-builtin-macro-redefined -D__DATE__=\"redacted\" -D__TIMESTAMP__=\"redacted\" -D__TIME__=\"redacted\"" ASMFLAGS="-U_FORTIFY_SOURCE -fstack-protector -Wall -Wthread-safety -Wself-assign -fcolor-diagnostics -fno-omit-frame-pointer -no-canonical-prefixes -Wno-builtin-macro-redefined -D__DATE__=\"redacted\" -D__TIMESTAMP__=\"redacted\" -D__TIME__=\"redacted\"" cmake -DCMAKE_AR="/usr/bin/ar" -DCMAKE_SHARED_LINKER_FLAGS="-shared -fuse-ld=gold -Wl,-no-as-needed -Wl,-z,relro,-z,now -lstdc++ -lm -fuse-ld=lld-7" -DCMAKE_EXE_LINKER_FLAGS="-fuse-ld=gold -Wl,-no-as-needed -Wl,-z,relro,-z,now -lstdc++ -lm -fuse-ld=lld-7" -DEVENT__DISABLE_OPENSSL="on" -DEVENT__DISABLE_REGRESS="on" -DCMAKE_BUILD_TYPE="Release" -DCMAKE_PREFIX_PATH="$EXT_BUILD_DEPS" -DCMAKE_INSTALL_PREFIX="$INSTALL_PREFIX" -GNinja $EXT_BUILD_ROOT/external/com_github_libevent_libevent
ninja
ninja install
cp -L -r --no-target-directory "$BUILD_TMPDIR/$INSTALL_PREFIX" "$INSTALLDIR" 
replace_in_files $INSTALLDIR $BUILD_TMPDIR BAZEL_GEN_ROOT
replace_in_files $INSTALLDIR $EXT_BUILD_DEPS BAZEL_GEN_ROOT
mkdir -p $EXT_BUILD_ROOT/bazel-out/k8-fastbuild/genfiles/bazel/foreign_cc/copy_event/event
cp -L -r --no-target-directory "$INSTALLDIR" "$EXT_BUILD_ROOT/bazel-out/k8-fastbuild/genfiles/bazel/foreign_cc/copy_event/event" 
touch $EXT_BUILD_ROOT/bazel-out/k8-fastbuild/genfiles/bazel/foreign_cc/empty_event.txt
cd $EXT_BUILD_ROOT
DEBUG: /home/mklein/.cache/bazel/_bazel_mklein/0eed830f48870bc411befee27d7940f5/external/foreign_cc_impl/framework.bzl:269:5: script text: function cleanup_function() {
  local ecode=$?
  if [ $ecode -eq 0 ]; then
  printf "$CLEANUP_MSG"
  rm -rf $BUILD_TMPDIR $EXT_BUILD_DEPS
  else
  printf ""
  printf "$KEEP_MSG"
  printf ""
  fi
}
function children_to_path() {
  if [ -d $EXT_BUILD_DEPS/bin ]; then
    local tools=$(find $EXT_BUILD_DEPS/bin -maxdepth 1 -mindepth 1)
    for tool in $tools;
    do
      if  [[ -d "$tool" ]] || [[ -L "$tool" ]]; then
        export PATH=$PATH:$tool
      fi
    done
  fi
}
function replace_in_files() {
  if [ -d "$1" ]; then
    find -L $1 -print -type f   \( -name "*.pc" -or -name "*.la" -or -name "*-config" -or -name "*.cmake" \)   -exec sed -i 's@'"$2"'@'"$3"'@g' {} ';'
  fi
}
printf ""
printf "Bazel external C/C++ Rules #0.0.6. Building library 'yaml'"
printf ""
set -e

export EXT_BUILD_ROOT=$(pwd)
export BUILD_TMPDIR=$(mktemp -d)
export EXT_BUILD_DEPS=$EXT_BUILD_ROOT/bazel_foreign_cc_deps_yaml
export INSTALLDIR=$EXT_BUILD_ROOT/bazel-out/k8-fastbuild/genfiles/bazel/foreign_cc/yaml
export PATH="$EXT_BUILD_ROOT:$PATH"
mkdir -p $EXT_BUILD_DEPS
mkdir -p $INSTALLDIR
printf "Environment:______________"
env
printf "__________________________"
export CLEANUP_MSG="rules_foreign_cc: Cleaning temp directories"
export KEEP_MSG="rules_foreign_cc: Keeping temp build directory $BUILD_TMPDIR and dependencies directory $EXT_BUILD_DEPS for debug.\nrules_foreign_cc: Please note that the directories inside a sandbox are still cleaned unless you specify '--sandbox_debug' Bazel command line flag."
trap "cleanup_function" EXIT
children_to_path $EXT_BUILD_DEPS/bin
export PATH="$EXT_BUILD_DEPS/bin:$PATH"
replace_in_files $EXT_BUILD_DEPS BAZEL_GEN_ROOT $EXT_BUILD_DEPS
cd $BUILD_TMPDIR
export INSTALL_PREFIX="./yaml"
CC="/home/mklein/.cache/bazel/_bazel_mklein/0eed830f48870bc411befee27d7940f5/external/local_config_cc/extra_tools/envoy_cc_wrapper" CXX="/home/mklein/.cache/bazel/_bazel_mklein/0eed830f48870bc411befee27d7940f5/external/local_config_cc/extra_tools/envoy_cc_wrapper" CFLAGS="-U_FORTIFY_SOURCE -fstack-protector -Wall -Wthread-safety -Wself-assign -fcolor-diagnostics -fno-omit-frame-pointer -no-canonical-prefixes -Wno-builtin-macro-redefined -D__DATE__=\"redacted\" -D__TIMESTAMP__=\"redacted\" -D__TIME__=\"redacted\"" CXXFLAGS="-U_FORTIFY_SOURCE -fstack-protector -Wall -Wthread-safety -Wself-assign -fcolor-diagnostics -fno-omit-frame-pointer -std=c++0x -no-canonical-prefixes -Wno-builtin-macro-redefined -D__DATE__=\"redacted\" -D__TIMESTAMP__=\"redacted\" -D__TIME__=\"redacted\"" ASMFLAGS="-U_FORTIFY_SOURCE -fstack-protector -Wall -Wthread-safety -Wself-assign -fcolor-diagnostics -fno-omit-frame-pointer -no-canonical-prefixes -Wno-builtin-macro-redefined -D__DATE__=\"redacted\" -D__TIMESTAMP__=\"redacted\" -D__TIME__=\"redacted\"" cmake -DCMAKE_AR="/usr/bin/ar" -DCMAKE_SHARED_LINKER_FLAGS="-shared -fuse-ld=gold -Wl,-no-as-needed -Wl,-z,relro,-z,now -lstdc++ -lm -fuse-ld=lld-7" -DCMAKE_EXE_LINKER_FLAGS="-fuse-ld=gold -Wl,-no-as-needed -Wl,-z,relro,-z,now -lstdc++ -lm -fuse-ld=lld-7" -DYAML_CPP_BUILD_TESTS="off" -DCMAKE_BUILD_TYPE="RelWithDebInfo" -DCMAKE_PREFIX_PATH="$EXT_BUILD_DEPS" -DCMAKE_INSTALL_PREFIX="$INSTALL_PREFIX" -GNinja $EXT_BUILD_ROOT/external/com_github_jbeder_yaml_cpp
ninja
ninja install
cp -L -r --no-target-directory "$BUILD_TMPDIR/$INSTALL_PREFIX" "$INSTALLDIR" 
replace_in_files $INSTALLDIR $BUILD_TMPDIR BAZEL_GEN_ROOT
replace_in_files $INSTALLDIR $EXT_BUILD_DEPS BAZEL_GEN_ROOT
mkdir -p $EXT_BUILD_ROOT/bazel-out/k8-fastbuild/genfiles/bazel/foreign_cc/copy_yaml/yaml
cp -L -r --no-target-directory "$INSTALLDIR" "$EXT_BUILD_ROOT/bazel-out/k8-fastbuild/genfiles/bazel/foreign_cc/copy_yaml/yaml" 
touch $EXT_BUILD_ROOT/bazel-out/k8-fastbuild/genfiles/bazel/foreign_cc/empty_yaml.txt
cd $EXT_BUILD_ROOT
DEBUG: /home/mklein/.cache/bazel/_bazel_mklein/0eed830f48870bc411befee27d7940f5/external/foreign_cc_impl/framework.bzl:269:5: script text: function cleanup_function() {
  local ecode=$?
  if [ $ecode -eq 0 ]; then
  printf "$CLEANUP_MSG"
  rm -rf $BUILD_TMPDIR $EXT_BUILD_DEPS
  else
  printf ""
  printf "$KEEP_MSG"
  printf ""
  fi
}
function children_to_path() {
  if [ -d $EXT_BUILD_DEPS/bin ]; then
    local tools=$(find $EXT_BUILD_DEPS/bin -maxdepth 1 -mindepth 1)
    for tool in $tools;
    do
      if  [[ -d "$tool" ]] || [[ -L "$tool" ]]; then
        export PATH=$PATH:$tool
      fi
    done
  fi
}
function replace_in_files() {
  if [ -d "$1" ]; then
    find -L $1 -print -type f   \( -name "*.pc" -or -name "*.la" -or -name "*-config" -or -name "*.cmake" \)   -exec sed -i 's@'"$2"'@'"$3"'@g' {} ';'
  fi
}
printf ""
printf "Bazel external C/C++ Rules #0.0.6. Building library 'ares'"
printf ""
set -e

export EXT_BUILD_ROOT=$(pwd)
export BUILD_TMPDIR=$(mktemp -d)
export EXT_BUILD_DEPS=$EXT_BUILD_ROOT/bazel_foreign_cc_deps_ares
export INSTALLDIR=$EXT_BUILD_ROOT/bazel-out/k8-fastbuild/genfiles/bazel/foreign_cc/ares
export PATH="$EXT_BUILD_ROOT:$PATH"
mkdir -p $EXT_BUILD_DEPS
mkdir -p $INSTALLDIR
printf "Environment:______________"
env
printf "__________________________"
export CLEANUP_MSG="rules_foreign_cc: Cleaning temp directories"
export KEEP_MSG="rules_foreign_cc: Keeping temp build directory $BUILD_TMPDIR and dependencies directory $EXT_BUILD_DEPS for debug.\nrules_foreign_cc: Please note that the directories inside a sandbox are still cleaned unless you specify '--sandbox_debug' Bazel command line flag."
trap "cleanup_function" EXIT
children_to_path $EXT_BUILD_DEPS/bin
export PATH="$EXT_BUILD_DEPS/bin:$PATH"
replace_in_files $EXT_BUILD_DEPS BAZEL_GEN_ROOT $EXT_BUILD_DEPS
cd $BUILD_TMPDIR
export INSTALL_PREFIX="./ares"
CC="/home/mklein/.cache/bazel/_bazel_mklein/0eed830f48870bc411befee27d7940f5/external/local_config_cc/extra_tools/envoy_cc_wrapper" CXX="/home/mklein/.cache/bazel/_bazel_mklein/0eed830f48870bc411befee27d7940f5/external/local_config_cc/extra_tools/envoy_cc_wrapper" CFLAGS="-U_FORTIFY_SOURCE -fstack-protector -Wall -Wthread-safety -Wself-assign -fcolor-diagnostics -fno-omit-frame-pointer -no-canonical-prefixes -Wno-builtin-macro-redefined -D__DATE__=\"redacted\" -D__TIMESTAMP__=\"redacted\" -D__TIME__=\"redacted\"" CXXFLAGS="-U_FORTIFY_SOURCE -fstack-protector -Wall -Wthread-safety -Wself-assign -fcolor-diagnostics -fno-omit-frame-pointer -std=c++0x -no-canonical-prefixes -Wno-builtin-macro-redefined -D__DATE__=\"redacted\" -D__TIMESTAMP__=\"redacted\" -D__TIME__=\"redacted\"" ASMFLAGS="-U_FORTIFY_SOURCE -fstack-protector -Wall -Wthread-safety -Wself-assign -fcolor-diagnostics -fno-omit-frame-pointer -no-canonical-prefixes -Wno-builtin-macro-redefined -D__DATE__=\"redacted\" -D__TIMESTAMP__=\"redacted\" -D__TIME__=\"redacted\"" cmake -DCMAKE_AR="/usr/bin/ar" -DCMAKE_SHARED_LINKER_FLAGS="-shared -fuse-ld=gold -Wl,-no-as-needed -Wl,-z,relro,-z,now -lstdc++ -lm -fuse-ld=lld-7" -DCMAKE_EXE_LINKER_FLAGS="-fuse-ld=gold -Wl,-no-as-needed -Wl,-z,relro,-z,now -lstdc++ -lm -fuse-ld=lld-7" -DCARES_SHARED="no" -DCARES_STATIC="on" -DCMAKE_BUILD_TYPE="RelWithDebInfo" -DCMAKE_PREFIX_PATH="$EXT_BUILD_DEPS" -DCMAKE_INSTALL_PREFIX="$INSTALL_PREFIX" -GNinja $EXT_BUILD_ROOT/external/com_github_c_ares_c_ares
ninja
ninja install
cp -L -r --no-target-directory "$BUILD_TMPDIR/$INSTALL_PREFIX" "$INSTALLDIR" 
replace_in_files $INSTALLDIR $BUILD_TMPDIR BAZEL_GEN_ROOT
replace_in_files $INSTALLDIR $EXT_BUILD_DEPS BAZEL_GEN_ROOT
mkdir -p $EXT_BUILD_ROOT/bazel-out/k8-fastbuild/genfiles/bazel/foreign_cc/copy_ares/ares
cp -L -r --no-target-directory "$INSTALLDIR" "$EXT_BUILD_ROOT/bazel-out/k8-fastbuild/genfiles/bazel/foreign_cc/copy_ares/ares" 
touch $EXT_BUILD_ROOT/bazel-out/k8-fastbuild/genfiles/bazel/foreign_cc/empty_ares.txt
cd $EXT_BUILD_ROOT
DEBUG: /home/mklein/.cache/bazel/_bazel_mklein/0eed830f48870bc411befee27d7940f5/external/foreign_cc_impl/framework.bzl:269:5: script text: function cleanup_function() {
  local ecode=$?
  if [ $ecode -eq 0 ]; then
  printf "$CLEANUP_MSG"
  rm -rf $BUILD_TMPDIR $EXT_BUILD_DEPS
  else
  printf ""
  printf "$KEEP_MSG"
  printf ""
  fi
}
function children_to_path() {
  if [ -d $EXT_BUILD_DEPS/bin ]; then
    local tools=$(find $EXT_BUILD_DEPS/bin -maxdepth 1 -mindepth 1)
    for tool in $tools;
    do
      if  [[ -d "$tool" ]] || [[ -L "$tool" ]]; then
        export PATH=$PATH:$tool
      fi
    done
  fi
}
function replace_in_files() {
  if [ -d "$1" ]; then
    find -L $1 -print -type f   \( -name "*.pc" -or -name "*.la" -or -name "*-config" -or -name "*.cmake" \)   -exec sed -i 's@'"$2"'@'"$3"'@g' {} ';'
  fi
}
printf ""
printf "Bazel external C/C++ Rules #0.0.6. Building library 'nghttp2'"
printf ""
set -e

export EXT_BUILD_ROOT=$(pwd)
export BUILD_TMPDIR=$(mktemp -d)
export EXT_BUILD_DEPS=$EXT_BUILD_ROOT/bazel_foreign_cc_deps_nghttp2
export INSTALLDIR=$EXT_BUILD_ROOT/bazel-out/k8-fastbuild/genfiles/bazel/foreign_cc/nghttp2
export PATH="$EXT_BUILD_ROOT:$PATH"
mkdir -p $EXT_BUILD_DEPS
mkdir -p $INSTALLDIR
printf "Environment:______________"
env
printf "__________________________"
export CLEANUP_MSG="rules_foreign_cc: Cleaning temp directories"
export KEEP_MSG="rules_foreign_cc: Keeping temp build directory $BUILD_TMPDIR and dependencies directory $EXT_BUILD_DEPS for debug.\nrules_foreign_cc: Please note that the directories inside a sandbox are still cleaned unless you specify '--sandbox_debug' Bazel command line flag."
trap "cleanup_function" EXIT
children_to_path $EXT_BUILD_DEPS/bin
export PATH="$EXT_BUILD_DEPS/bin:$PATH"
replace_in_files $EXT_BUILD_DEPS BAZEL_GEN_ROOT $EXT_BUILD_DEPS
cd $BUILD_TMPDIR
export INSTALL_PREFIX="./nghttp2"
CC="/home/mklein/.cache/bazel/_bazel_mklein/0eed830f48870bc411befee27d7940f5/external/local_config_cc/extra_tools/envoy_cc_wrapper" CXX="/home/mklein/.cache/bazel/_bazel_mklein/0eed830f48870bc411befee27d7940f5/external/local_config_cc/extra_tools/envoy_cc_wrapper" CFLAGS="-U_FORTIFY_SOURCE -fstack-protector -Wall -Wthread-safety -Wself-assign -fcolor-diagnostics -fno-omit-frame-pointer -no-canonical-prefixes -Wno-builtin-macro-redefined -D__DATE__=\"redacted\" -D__TIMESTAMP__=\"redacted\" -D__TIME__=\"redacted\"" CXXFLAGS="-U_FORTIFY_SOURCE -fstack-protector -Wall -Wthread-safety -Wself-assign -fcolor-diagnostics -fno-omit-frame-pointer -std=c++0x -no-canonical-prefixes -Wno-builtin-macro-redefined -D__DATE__=\"redacted\" -D__TIMESTAMP__=\"redacted\" -D__TIME__=\"redacted\"" ASMFLAGS="-U_FORTIFY_SOURCE -fstack-protector -Wall -Wthread-safety -Wself-assign -fcolor-diagnostics -fno-omit-frame-pointer -no-canonical-prefixes -Wno-builtin-macro-redefined -D__DATE__=\"redacted\" -D__TIMESTAMP__=\"redacted\" -D__TIME__=\"redacted\"" cmake -DCMAKE_AR="/usr/bin/ar" -DCMAKE_SHARED_LINKER_FLAGS="-shared -fuse-ld=gold -Wl,-no-as-needed -Wl,-z,relro,-z,now -lstdc++ -lm -fuse-ld=lld-7" -DCMAKE_EXE_LINKER_FLAGS="-fuse-ld=gold -Wl,-no-as-needed -Wl,-z,relro,-z,now -lstdc++ -lm -fuse-ld=lld-7" -DENABLE_STATIC_LIB="on" -DENABLE_LIB_ONLY="on" -DCMAKE_BUILD_TYPE="RelWithDebInfo" -DCMAKE_PREFIX_PATH="$EXT_BUILD_DEPS" -DCMAKE_INSTALL_PREFIX="$INSTALL_PREFIX" -GNinja $EXT_BUILD_ROOT/external/com_github_nghttp2_nghttp2
ninja
ninja install
cp -L -r --no-target-directory "$BUILD_TMPDIR/$INSTALL_PREFIX" "$INSTALLDIR" 
replace_in_files $INSTALLDIR $BUILD_TMPDIR BAZEL_GEN_ROOT
replace_in_files $INSTALLDIR $EXT_BUILD_DEPS BAZEL_GEN_ROOT
mkdir -p $EXT_BUILD_ROOT/bazel-out/k8-fastbuild/genfiles/bazel/foreign_cc/copy_nghttp2/nghttp2
cp -L -r --no-target-directory "$INSTALLDIR" "$EXT_BUILD_ROOT/bazel-out/k8-fastbuild/genfiles/bazel/foreign_cc/copy_nghttp2/nghttp2" 
touch $EXT_BUILD_ROOT/bazel-out/k8-fastbuild/genfiles/bazel/foreign_cc/empty_nghttp2.txt
cd $EXT_BUILD_ROOT
DEBUG: /home/mklein/.cache/bazel/_bazel_mklein/0eed830f48870bc411befee27d7940f5/external/foreign_cc_impl/framework.bzl:269:5: script text: function cleanup_function() {
  local ecode=$?
  if [ $ecode -eq 0 ]; then
  printf "$CLEANUP_MSG"
  rm -rf $BUILD_TMPDIR $EXT_BUILD_DEPS
  else
  printf ""
  printf "$KEEP_MSG"
  printf ""
  fi
}
function children_to_path() {
  if [ -d $EXT_BUILD_DEPS/bin ]; then
    local tools=$(find $EXT_BUILD_DEPS/bin -maxdepth 1 -mindepth 1)
    for tool in $tools;
    do
      if  [[ -d "$tool" ]] || [[ -L "$tool" ]]; then
        export PATH=$PATH:$tool
      fi
    done
  fi
}
function replace_in_files() {
  if [ -d "$1" ]; then
    find -L $1 -print -type f   \( -name "*.pc" -or -name "*.la" -or -name "*-config" -or -name "*.cmake" \)   -exec sed -i 's@'"$2"'@'"$3"'@g' {} ';'
  fi
}
printf ""
printf "Bazel external C/C++ Rules #0.0.6. Building library 'zlib'"
printf ""
set -e

export EXT_BUILD_ROOT=$(pwd)
export BUILD_TMPDIR=$(mktemp -d)
export EXT_BUILD_DEPS=$EXT_BUILD_ROOT/bazel_foreign_cc_deps_zlib
export INSTALLDIR=$EXT_BUILD_ROOT/bazel-out/k8-fastbuild/genfiles/bazel/foreign_cc/zlib
export PATH="$EXT_BUILD_ROOT:$PATH"
mkdir -p $EXT_BUILD_DEPS
mkdir -p $INSTALLDIR
printf "Environment:______________"
env
printf "__________________________"
export CLEANUP_MSG="rules_foreign_cc: Cleaning temp directories"
export KEEP_MSG="rules_foreign_cc: Keeping temp build directory $BUILD_TMPDIR and dependencies directory $EXT_BUILD_DEPS for debug.\nrules_foreign_cc: Please note that the directories inside a sandbox are still cleaned unless you specify '--sandbox_debug' Bazel command line flag."
trap "cleanup_function" EXIT
children_to_path $EXT_BUILD_DEPS/bin
export PATH="$EXT_BUILD_DEPS/bin:$PATH"
replace_in_files $EXT_BUILD_DEPS BAZEL_GEN_ROOT $EXT_BUILD_DEPS
cd $BUILD_TMPDIR
export INSTALL_PREFIX="./zlib"
CC="/home/mklein/.cache/bazel/_bazel_mklein/0eed830f48870bc411befee27d7940f5/external/local_config_cc/extra_tools/envoy_cc_wrapper" CXX="/home/mklein/.cache/bazel/_bazel_mklein/0eed830f48870bc411befee27d7940f5/external/local_config_cc/extra_tools/envoy_cc_wrapper" CFLAGS="-U_FORTIFY_SOURCE -fstack-protector -Wall -Wthread-safety -Wself-assign -fcolor-diagnostics -fno-omit-frame-pointer -no-canonical-prefixes -Wno-builtin-macro-redefined -D__DATE__=\"redacted\" -D__TIMESTAMP__=\"redacted\" -D__TIME__=\"redacted\"" CXXFLAGS="-U_FORTIFY_SOURCE -fstack-protector -Wall -Wthread-safety -Wself-assign -fcolor-diagnostics -fno-omit-frame-pointer -std=c++0x -no-canonical-prefixes -Wno-builtin-macro-redefined -D__DATE__=\"redacted\" -D__TIMESTAMP__=\"redacted\" -D__TIME__=\"redacted\"" ASMFLAGS="-U_FORTIFY_SOURCE -fstack-protector -Wall -Wthread-safety -Wself-assign -fcolor-diagnostics -fno-omit-frame-pointer -no-canonical-prefixes -Wno-builtin-macro-redefined -D__DATE__=\"redacted\" -D__TIMESTAMP__=\"redacted\" -D__TIME__=\"redacted\"" cmake -DCMAKE_AR="/usr/bin/ar" -DCMAKE_SHARED_LINKER_FLAGS="-shared -fuse-ld=gold -Wl,-no-as-needed -Wl,-z,relro,-z,now -lstdc++ -lm -fuse-ld=lld-7" -DCMAKE_EXE_LINKER_FLAGS="-fuse-ld=gold -Wl,-no-as-needed -Wl,-z,relro,-z,now -lstdc++ -lm -fuse-ld=lld-7" -DCMAKE_BUILD_TYPE="RelWithDebInfo" -DCMAKE_PREFIX_PATH="$EXT_BUILD_DEPS" -DCMAKE_INSTALL_PREFIX="$INSTALL_PREFIX" -GNinja $EXT_BUILD_ROOT/external/com_github_madler_zlib
ninja
ninja install
cp -L -r --no-target-directory "$BUILD_TMPDIR/$INSTALL_PREFIX" "$INSTALLDIR" 
replace_in_files $INSTALLDIR $BUILD_TMPDIR BAZEL_GEN_ROOT
replace_in_files $INSTALLDIR $EXT_BUILD_DEPS BAZEL_GEN_ROOT
mkdir -p $EXT_BUILD_ROOT/bazel-out/k8-fastbuild/genfiles/bazel/foreign_cc/copy_zlib/zlib
cp -L -r --no-target-directory "$INSTALLDIR" "$EXT_BUILD_ROOT/bazel-out/k8-fastbuild/genfiles/bazel/foreign_cc/copy_zlib/zlib" 
touch $EXT_BUILD_ROOT/bazel-out/k8-fastbuild/genfiles/bazel/foreign_cc/empty_zlib.txt
cd $EXT_BUILD_ROOT
DEBUG: /home/mklein/Source/envoy/bazel/repositories.bzl:89:5: Fetching external dependencies...
DEBUG: /home/mklein/Source/envoy/bazel/repositories.bzl:95:5: 
DEBUG: /home/mklein/Source/envoy/bazel/repositories.bzl:96:5: External dependency cache directory /home/mklein/.cache/bazel/_bazel_mklein/0eed830f48870bc411befee27d7940f5/external/envoy_deps_cache_98bb5acb9d3b3dd417aa7cbace7d01e9
make: Entering directory '/home/mklein/.cache/bazel/_bazel_mklein/0eed830f48870bc411befee27d7940f5/external/envoy_deps'
No need to rebuild /home/mklein/.cache/bazel/_bazel_mklein/0eed830f48870bc411befee27d7940f5/external/envoy_deps_cache_98bb5acb9d3b3dd417aa7cbace7d01e9/gperftools.dep.env
No need to rebuild /home/mklein/.cache/bazel/_bazel_mklein/0eed830f48870bc411befee27d7940f5/external/envoy_deps_cache_98bb5acb9d3b3dd417aa7cbace7d01e9/luajit.dep.env
make: Leaving directory '/home/mklein/.cache/bazel/_bazel_mklein/0eed830f48870bc411befee27d7940f5/external/envoy_deps'

real	0m0.112s
user	0m0.012s
sys	0m0.001s
DEBUG: /home/mklein/Source/envoy/bazel/repositories.bzl:97:5: External dep build exited with return code: 0
INFO: Analysed 9 targets (0 packages loaded, 0 targets configured).
INFO: Found 9 targets...
[0 / 1] [-----] BazelWorkspaceStatusAction stable-status.txt
INFO: Elapsed time: 0.216s, Critical Path: 0.04s
INFO: 0 processes.
INFO: Build completed successfully, 1 total action
INFO: Build completed successfully, 1 total action
[SE mklein@localhost envoy]$ 
```

**Comment by irengrig** [comment]
Please have a look at the situation after #223; almost all output now is redirected to log files, and only printed to standard output in case of error.

**Comment by irengrig** [comment]
Closing as it was fixed by #223
